Hivemind/Assets/Scripts/Gathering.cs

```diff
@@ -143,4 +210,61 @@ private void Update()
                 break;
         }
     }
-}
+
+    private IEnumerator Scout()
+    {
+        Vector3 destination = new Vector3(transform.position.x + Random.Range(-5, 5), transform.position.y, transform.position.z + Random.Range(-5, 5));
+        switch (PrefferedDirection)
+        {
+            case Direction.None:
+                break;
+
+            case Direction.North:
+                destination.z += 2;
+                break;
+
+            case Direction.West:
+                destination.x -= 2;
+                break;
+
+            case Direction.South:
+                destination.z -= 2;
+                break;
+
+            case Direction.East:
+                destination.x += 2;
+                break;
+
+            case Direction.NorthWest:
+                destination.z += 2;
+                destination.x -= 2;
+                break;
+
+            case Direction.NorthEast:
+                destination.z += 2;
+                destination.x += 2;
+                break;
+
+            case Direction.SouthEast:
+                destination.z -= 2;
+                destination.x += 2;
+                break;
+
+            case Direction.SouthWest:
+                destination.z -= 2;
+                destination.x -= 2;
+                break;
+        }
+        yield return new WaitForSeconds(Random.Range(1, 3));
+        agent.SetDestination(destination);
+        scouting = false;
+    }
+
+    private IEnumerator ReturnToBase()
+    {
+        yield return new WaitForSeconds(Random.Range(30, 60));
```

Comment:
ik snap dit niet zo goed waarom wachten we 30 tot 60 seconden met het terug lopen?

Reply:
Als na 30-60 seconden scouten niks is gevonden gaan ze terug naar base, zo blijven ze niet buiten tot ze iets vinden of dood gaan